client/client.go

```diff
@@ -5,21 +5,58 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
 	"github.com/hashicorp/go-cleanhttp"
 	"github.com/hashicorp/go-retryablehttp"
 	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
 )
 
+// last retry will wait 64 seconds (2^6)
+// will also listen to 'Retry-After' header
+
+const (
+	rpm              = 600
```

Comment:
I think our HTTP response includes details about when to retry. We don't set the HTTP headers (yet, though you could opt to do this here) but you can also just read the response and use them to power your retries if you want.

It's likely not a good idea to add a different rate-limit construct on here on top of our rate-limit. We can just reduce the default parallelism of the work to keep it below the expected rate-limit rather than pessimistically rate-limit in advance, I think?

Reply:
I've changed behaviour to: check the header, if not check the body, if not do the stupid backoff